deployment/cloud/aws/account/main.tf

```diff
@@ -0,0 +1,177 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.61"
+    }
+  }
+
+  backend "s3" {
+    bucket = "${var.account_resource_prefix}-${var.account_id}-opentofu-state"
+    key = "account/terraform.tfstate"
+    region = "us-west-2"
+  }
+}
+
+provider "aws" {
+  region = "us-west-2"
+
+  // Tags all resources created from this provider with {"Project": <project name input>, "Neighborhood": <neighborhood input>}
+  // as well as any additional tags provided
+  default_tags {
+    tags = merge(
+      var.account_additional_tags
+    )
+  }
+}
+
+# s3 tf state bucket
+
+resource "aws_s3_bucket" "tf_state" {
+  bucket = "${var.account_resource_prefix}-${var.account_id}-opentofu-state"
+}
+
+resource "aws_s3_bucket_versioning" "this" {
+  bucket = aws_s3_bucket.tf_state.bucket
+
+  versioning_configuration {
+    status = "Enabled"
+  }
+  
+}
+
+resource "aws_s3_bucket_public_access_block" "example" {
+  bucket = aws_s3_bucket.tf_state.bucket
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+# deploy role
+resource "aws_iam_role" "deploy" {
+  name = "${var.account_resource_prefix}-deploy"
+  description = "Role used to deploy infrastructure for the Support Sphere app, part of the Post-Disaster communications project."
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Principal = {
+          AWS = "arn:aws:iam::${var.account_id}:root"
+        },
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+
+  managed_policy_arns = [
+    "arn:aws:iam::aws:policy/AmazonEC2FullAccess",
+    "arn:aws:iam::aws:policy/AmazonVPCFullAccess",
+    "arn:aws:iam::aws:policy/AutoScalingFullAccess",
+    "arn:aws:iam::aws:policy/IAMFullAccess",
```

Comment:
Does this deploy user need an IAM Full Access? They're not messing around with IAM at the deployment stage right?

Reply:
They need to create ops roles & associate their permissions with the `ssec-eng` group, instance policies, policies encrypt/decrypt access for the SOPS keys, etc. This could definitely be scoped down (which I do with some `deny` roles on all the IAM User-related-stuff further down here)

tl;dr -- they need some IAM access at the deploy stage